### ISSUE TITLE
Implement multi-period analysis capabilities

### ISSUE THREAD

**Issue by GGcarlson** [GGcarlson/dashboard-script#9]
## Problem
The current dashboard only compares current period vs previous period, limiting analysis capabilities. Small businesses need to analyze trends across multiple time periods (quarterly, yearly) and perform historical comparisons for better financial planning and decision-making.

## Technical Requirements
- Support analysis across multiple time periods (monthly, quarterly, yearly)
- Implement rolling averages and trend calculations
- Add year-over-year comparisons
- Create historical data analysis with forecasting
- Support flexible date range selection
- Generate comparative reports across custom periods

## Acceptance Criteria
- [ ] Multi-period P&L statements (3-year, 5-year comparisons)
- [ ] Quarterly and yearly trend analysis
- [ ] Rolling 12-month calculations
- [ ] Year-over-year growth metrics
- [ ] Seasonal analysis and patterns
- [ ] Custom date range comparisons
- [ ] Forecasting based on historical trends
- [ ] Budget vs actual across multiple periods
- [ ] Variance analysis over time
- [ ] Configurable fiscal year support

### 1. Period Configuration System
```python
class PeriodAnalyzer:
    def __init__(self, transaction_data):
        self.data = transaction_data
        self.fiscal_year_start = 1  # January default
    
    def get_period_data(self, period_type, num_periods=12):
        \"\"\"Get data for specified number of periods\"\"\"
        periods = {
            'monthly': self.get_monthly_periods(num_periods),
            'quarterly': self.get_quarterly_periods(num_periods),
            'yearly': self.get_yearly_periods(num_periods)
        }
        return periods.get(period_type, [])
    
    def calculate_rolling_average(self, metric, window_months=12):
        \"\"\"Calculate rolling average for any metric\"\"\"
        monthly_data = self.get_monthly_data()
        return monthly_data[metric].rolling(window=window_months).mean()
    
    def year_over_year_growth(self, metric):
        \"\"\"Calculate YoY growth rates\"\"\"
        yearly_data = self.get_yearly_data()
        return yearly_data[metric].pct_change() * 100
```

### 2. Multi-Period Dashboard Layout
```
┌─ Financial Performance Analysis (2022-2024) ────────────────────────────┐
│                                                                          │
│ ┌─ 3-Year P&L Comparison ──────────────────────────────────────────────┐ │
│ │                    2022        2023        2024      3-Yr Avg       │ │
│ │ Revenue           $245,000     $267,000    $289,000   $267,000       │ │
│ │ Gross Profit      $180,000     $195,000    $210,000   $195,000       │ │
│ │ Net Income        $45,000      $52,000     $58,000    $51,667        │ │
│ └──────────────────────────────────────────────────────────────────────┘ │
│                                                                          │
│ ┌─ Quarterly Trends (2024) ────────────────────────────────────────────┐ │
│ │           Q1        Q2        Q3        Q4*       YTD               │ │
│ │ Revenue   $68,000   $72,000   $75,000   $74,000   $289,000          │ │
│ │ Growth    +8.2%     +5.9%     +4.2%     -1.3%     +8.2%            │ │
│ └──────────────────────────────────────────────────────────────────────┘ │
│                                                                          │
│ ┌─ Key Metrics & Trends ───────────────────────────────────────────────┐ │
│ │ Revenue Growth (YoY):      8.2%   │  Profit Margin Trend: ↗        │ │
│ │ Rolling 12-Mo Revenue:  $285,000   │  Seasonal Peak:     Q2-Q3      │ │
│ │ Best Month:          June 2024     │  Growth Rate:       +6.8%      │ │
│ └──────────────────────────────────────────────────────────────────────┘ │
└──────────────────────────────────────────────────────────────────────────┘
```

### 3. Excel Formula Implementation
```python
def create_multi_period_formulas(self, ws):
    \"\"\"Create formulas for multi-period analysis\"\"\"
    
    # 3-Year revenue comparison
    for year_offset in range(3):
        year_cell = f'G{6 + year_offset}'
        start_date = f'DATE(YEAR(TODAY())-{2-year_offset},1,1)'
        end_date = f'DATE(YEAR(TODAY())-{2-year_offset},12,31)'
        
        revenue_formula = f'=SUMIFS(TransactionData[Amount], TransactionData[Type], \"Revenue\", TransactionData[Date], \">=\"&{start_date}, TransactionData[Date], \"<=\"&{end_date})'
        ws.range(year_cell).formula = revenue_formula
    
    # Rolling 12-month calculation
    rolling_cell = 'G10'
    rolling_formula = '=SUMIFS(TransactionData[Amount], TransactionData[Type], \"Revenue\", TransactionData[Date], \">=\"&DATE(YEAR(TODAY()),MONTH(TODAY())-11,1), TransactionData[Date], \"<=\"&TODAY())'
    ws.range(rolling_cell).formula = rolling_formula
    
    # Year-over-year growth
    growth_cell = 'G11'
    growth_formula = '=(G6-G7)/ABS(G7)'  # Current year vs previous year
    ws.range(growth_cell).formula = growth_formula
    ws.range(growth_cell).number_format = '0.0%'
```

### 1. Trend Analysis
```python
def calculate_trends(self, data_series):
    \"\"\"Calculate trend direction and strength\"\"\"
    from scipy import stats
    
    x = range(len(data_series))
    slope, intercept, r_value, p_value, std_err = stats.linregress(x, data_series)
    
    trend_strength = abs(r_value)
    trend_direction = 'increasing' if slope > 0 else 'decreasing'
    
    return {
        'direction': trend_direction,
        'strength': trend_strength,
        'slope': slope,
        'r_squared': r_value ** 2
    }
```

### 2. Seasonal Analysis
```python
def analyze_seasonality(self, monthly_data):
    \"\"\"Identify seasonal patterns\"\"\"
    seasonal_index = {}
    
    for month in range(1, 13):
        month_data = monthly_data[monthly_data['month'] == month]
        monthly_avg = month_data['amount'].mean()
        overall_avg = monthly_data['amount'].mean()
        
        seasonal_index[month] = monthly_avg / overall_avg
    
    return seasonal_index

def identify_peak_seasons(self, seasonal_index):
    \"\"\"Identify high and low seasons\"\"\"
    sorted_months = sorted(seasonal_index.items(), key=lambda x: x[1], reverse=True)
    
    peak_season = sorted_months[:3]  # Top 3 months
    low_season = sorted_months[-3:]  # Bottom 3 months
    
    return peak_season, low_season
```

### 3. Forecasting
```python
def simple_forecast(self, historical_data, periods_ahead=4):
    \"\"\"Generate simple forecast based on trends\"\"\"
    from sklearn.linear_model import LinearRegression
    import numpy as np
    
    # Prepare data for linear regression
    X = np.array(range(len(historical_data))).reshape(-1, 1)
    y = np.array(historical_data)
    
    # Fit model and predict
    model = LinearRegression()
    model.fit(X, y)
    
    # Forecast future periods
    future_X = np.array(range(len(historical_data), len(historical_data) + periods_ahead)).reshape(-1, 1)
    forecast = model.predict(future_X)
    
    # Calculate confidence intervals (simple approach)
    residuals = y - model.predict(X)
    std_residuals = np.std(residuals)
    
    return {
        'forecast': forecast,
        'confidence_interval': std_residuals * 1.96,  # 95% CI
        'trend_slope': model.coef_[0]
    }
```

### 1. Multi-Period Controls
```python
def add_period_controls(self, ws):
    \"\"\"Add dropdown controls for period selection\"\"\"
    
    # Analysis period dropdown
    period_options = ['Monthly', 'Quarterly', 'Yearly']
    ws.range('B2').value = 'Analysis Period:'
    ws.range('C2').validation.add(
        validation_type='list',
        validation_criteria='Monthly,Quarterly,Yearly'
    )
    
    # Number of periods
    ws.range('B3').value = 'Periods to Show:'
    ws.range('C3').value = 12
    
    # Fiscal year start
    ws.range('B4').value = 'Fiscal Year Starts:'
    ws.range('C4').validation.add(
        validation_type='list',
        validation_criteria='January,February,March,April,May,June,July,August,September,October,November,December'
    )
```

### 2. Dynamic Period Tables
```python
def create_dynamic_period_table(self, ws, start_row=10):
    \"\"\"Create table that adapts to selected period type\"\"\"
    
    # Headers adapt based on period selection
    period_type_cell = 'C2'
    
    # Use IF formulas to show appropriate headers
    ws.range(f'A{start_row}').formula = f'=IF({period_type_cell}=\"Monthly\",\"Month\",IF({period_type_cell}=\"Quarterly\",\"Quarter\",\"Year\"))'
    ws.range(f'B{start_row}').value = 'Revenue'
    ws.range(f'C{start_row}').value = 'Expenses'
    ws.range(f'D{start_row}').value = 'Net Income'
    ws.range(f'E{start_row}').value = 'Growth %'
    
    # Data rows with dynamic formulas based on period type
    for i in range(1, 13):  # Up to 12 periods
        row = start_row + i
        
        # Period formulas adapt to selection
        period_formula = f'=IF($C$2=\"Monthly\",DATE(YEAR(TODAY()),MONTH(TODAY())-{12-i},1),IF($C$2=\"Quarterly\",\"Q\"&ROUNDUP((MONTH(TODAY())-{4-i})/3,0),YEAR(TODAY())-{3-i}))'
        ws.range(f'A{row}').formula = period_formula
```

## Configuration Options
```yaml
multi_period_analysis:
  default_periods: 12
  fiscal_year_start: 1  # January
  analysis_types:
    - monthly
    - quarterly  
    - yearly
  
  forecasting:
    enabled: true
    method: 'linear_regression'
    periods_ahead: 4
    confidence_level: 0.95
  
  seasonal_analysis:
    enabled: true
    minimum_years: 2
```

## Priority
Medium - Provides advanced analytical capabilities for business planning

## Labels
enhancement, analytics, multi-period, forecasting, medium-priority

**Comment by GGcarlson** [comment]
Closing as part of repository cleanup. Focusing on core modular architecture.
